src/Neo/Persistence/DataCache.cs

```diff
@@ -72,6 +77,16 @@ public StorageItem this[StorageKey key]
             }
         }
 
+        /// <summary>
+        /// Data cache constructor
+        /// </summary>
+        /// <param name="readOnly">True if you don't want to allow writes</param>
+        protected DataCache(bool readOnly)
+        {
+            if (!readOnly)
+                _changeSet = [];
```

Comment:
is this the same as _changeSet = null; ?

cause you have ` public bool IsReadOnly => _changeSet == null;`